src/sdk/src/indexer_sdk.rs

```diff
@@ -0,0 +1,326 @@
+//
+// bill_sdk.rs
+// Copyright (C) 2022 db3.network Author imotai <codego.me@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use bytes::BytesMut;
+use chrono::Utc;
+use db3_crypto::{db3_address::DB3Address, db3_signer::Db3MultiSchemeSigner};
+use db3_proto::db3_account_proto::Account;
+use db3_proto::db3_database_proto::structured_query::{Limit, Projection};
+use db3_proto::db3_database_proto::{Database, Document, StructuredQuery};
+use db3_proto::db3_indexer_proto::{
+    indexer_node_client::IndexerNodeClient, GetDocumentRequest, IndexerStatus, RunQueryRequest,
+    RunQueryResponse, ShowDatabaseRequest, ShowIndexerStatusRequest,
+};
+use ethers::core::types::{
+    transaction::eip712::{EIP712Domain, TypedData, Types},
+    Bytes,
+};
+
+use hex;
+use num_traits::cast::FromPrimitive;
+use prost::Message;
+use std::collections::BTreeMap;
+use std::sync::atomic::AtomicBool;
+use std::sync::Arc;
+use tonic::Status;
+use uuid::Uuid;
+
+pub struct IndexerSDK {
+    client: Arc<IndexerNodeClient<tonic::transport::Channel>>,
+    signer: Db3MultiSchemeSigner,
+    types: Types,
+    use_typed_format: bool,
```

Comment:
you can delete the types and use_typed_format directly!

Reply:
Got it. will update it the next revision

Reply:
done